applications/asset_tracker_v2/src/modules/util_module.c

```diff
@@ -199,127 +203,133 @@ static void send_reboot_request(void)
 
 		EVENT_SUBMIT(util_module_event);
 
+		state_set(STATE_REBOOT_PENDING);
+
 		error_signaled = true;
 	}
 }
 
-/* Message handler for all states. */
-static void on_all_states(struct util_msg_data *msg)
+/* This API should be called exactly once for each _SHUTDOWN_READY event received from active
+ * modules in the application. When this API has been called a set number of times equal to the
+ * number of active modules, a reboot will be schedueled.
+ */
+static void reboot_ack_check(void)
 {
-	static int reboot_ack_cnt;
-
-	if (is_cloud_module_event(&msg->module.cloud.header)) {
-		switch (msg->module.cloud.type) {
-		case CLOUD_EVT_ERROR:
-			/* Fall through. */
-		case CLOUD_EVT_FOTA_DONE:
-			send_reboot_request();
-			break;
-		case CLOUD_EVT_SHUTDOWN_READY:
-			reboot_ack_cnt++;
-			break;
-		default:
-			break;
-		}
+	static uint8_t reboot_ack_count;
+
+	reboot_ack_count++;
```

Comment:
There is no protection against a module calling this multiple times, which could lead to an early reboot.

Reply:
This function is only visible to the utility module and can only be called in the ```REBOOT_PENDING``` state when the util module already expects shutdown acknowledgements from the other modules.